### PR TITLE
fix(bridge): stop_at_layer never stops on native Bridge

### LINKED ISSUE #1632: [Bug Report] Native TransformerBridge stop_at_layer never stops, causing input_to_embed() to return logits
**Describe the bug**

`TransformerBridge.boot_native()` does not honor `stop_at_layer` in a direct forward pass. As a consequence, `input_to_embed()` returns final vocabulary logits instead of the residual stream entering block 0.

This is silent: the calls succeed, but every requested stop point produces the full forward output. If `d_vocab == d_model`, even a shape check cannot distinguish the logits from the promised residual stream.

The root cause is `BlockBridge._extract_layer_idx()`. Native Bridge blocks are named `layers.0`, `layers.1`, and so on, while the parser only recognizes `\.layers.N` with a preceding dot:

```python
re.search(r"blocks\.(\d+)", self.name)
or re.search(r"\.h\.(\d+)", self.name)
or re.search(r"\.layers\.(\d+)", self.name)
```

For a top-level name such as `layers.0`, layer extraction returns `None`, so `_check_stop_at_layer()` never raises `StopAtLayerException`.

**Code example**

This reproduction is fully offline and uses a randomly initialized native Bridge:

```python
import torch

from transformer_lens.config import TransformerBridgeConfig
from transformer_lens.model_bridge import TransformerBridge

cfg = TransformerBridgeConfig(
    d_model=32,
    d_head=16,
    n_heads=2,
    n_layers=3,
    n_ctx=8,
    d_vocab=16,
    d_mlp=64,
    act_fn="gelu",
    normalization_type="LN",
    seed=0,
)
bridge = TransformerBridge.boot_native(cfg)
tokens = torch.tensor([[1, 2, 3]])

print([block.name for block in bridge.blocks])
for stop_at_layer in (0, 2, -1):
    print(stop_at_layer, tuple(bridge(tokens, stop_at_layer=stop_at_layer).shape))

residual, _, _, _ = bridge.input_to_embed(tokens)
print("input_to_embed", tuple(residual.shape))
```

Observed:

```text
['layers.0', 'layers.1', 'layers.2']
0  (1, 3, 16)
2  (1, 3, 16)
-1 (1, 3, 16)
input_to_embed (1, 3, 16)
```

All four results have the vocabulary dimension (`d_vocab=16`), showing that they are final logits. `stop_at_layer=0` and `input_to_embed()` should instead return the pre-block residual with shape `(1, 3, 32)`; other stop points should likewise return an intermediate residual with final dimension `d_model=32`.

`run_with_cache(tokens, stop_at_layer=...)` can appear correct because it installs a separate stop hook on `blocks.{k}.hook_in`. That path bypasses the broken block-name parser, so existing cache tests do not cover direct `forward(..., stop_at_layer=...)` or the `input_to_embed()` helper that depends on it.

**System Info**

* Installed from source on current `dev-4.x`
* Windows 11, CPU
* Python 3.12.10
* Reproduction uses `TransformerBridge.boot_native()` and requires no model download

**Additional context**

A boundary-aware parser such as `(?:^|\.)layers\.(\d+)` would recognize both top-level native names (`layers.0`) and nested HF-style names (`model.layers.0`). The `h` pattern should use the same boundary form so a future top-level `h.0` layout does not reproduce the bug.

Regression coverage should directly exercise `boot_native()` rather than relying only on `run_with_cache()`:

* direct `forward(..., stop_at_layer=0)`
* a positive intermediate layer and a negative layer index
* `input_to_embed()` returning a `d_model` residual
* resuming the returned residual with `start_at_layer=0` and matching a complete forward pass

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/TransformerLensOrg/TransformerLens/issues) in the repo (**required**)

### PR DESCRIPTION
**Pull request by LightWork666**
Fixes #1632

## What was broken

`stop_at_layer` silently did nothing on native Bridge — every stop value returned full logits instead of the residual stream. Two root causes:

**1. Regex mismatch in `_extract_layer_idx`** (`block.py`)

Native Bridge blocks are named `layers.0`, `layers.1`, etc. (top-level). The existing pattern `\.layers\.(\d+)` requires a preceding dot, so it matched `model.layers.0` (LLaMA-style) but returned `None` for `layers.0`. Since `_check_stop_at_layer` only fires when the index is known, the exception was never raised.

**2. Negative index stored raw** (`transformer_bridge.py`)

`block._stop_at_layer_idx = stop_at_layer` stored `-1` directly. `_check_stop_at_layer` then compared `layer_idx == -1` (e.g. `2 == -1`) which never matched. `bridge_core.py` already normalizes negative indices in its own stop path — this brings `transformer_bridge.py` in line.

## Changes

- `block.py`: `r"\.layers\.(\d+)"` → `r"(?:^|\.)layers\.(\d+)"` — matches both top-level and nested naming
- `transformer_bridge.py`: normalize negative `stop_at_layer` before stamping it on blocks
- `tests/unit/model_bridge/test_native_stop_at_layer.py`: 7 new unit tests covering stop values 0, 1, 2, -1, value-distinctness between stops, and `input_to_embed` returning `d_model` not `d_vocab`

## Test plan

- [ ] `uv run pytest tests/unit/model_bridge/test_native_stop_at_layer.py -v` — 7/7 pass
- [ ] `uv run pytest tests/unit/model_bridge/generalized_components/ -q` — 184/184 pass
- [ ] `uv run pytest tests/unit/model_bridge/ -q -k native` — all pass

### CODE REVIEW

**Comment by jlarson4** [comment]
@LightWork666 I am going to close this, but thank you for taking a look at it! #1633 was already filed in response to #1632  by the issues creator. It passed my review without any comments and is merged now.
